### DISCUSSION TITLE
Is there a way to trigger the reconnection of all the metamask accounts with the given snap?

### DISCUSSION THREAD

**Discussion by himalayan-dev** [MetaMask/snaps#2027] [General]
I have the following scenario:
1. User goes on the website
2. Website tries to install the snap
3. User accepts everything and the snap is installed
4. Website calls some snap API
5. User is asked to allow their metamask accounts to connect to the snap

This works well when the initial condition is where the snap isn't installed.

However, if the snap is already installed, there's currently no way to force user to re-request snap permissions. For SIWE, we can easily just call the "wallet_requestPermissions" API however, there's no similar function for the snap. 

Sometimes, there will be a case whereby user adds a new metamask account that was previously not authorized. We basically need a way to force user to re-authorize this newly added account because otherwise, the snap still uses the last connected metamask account if the current connected metamask account is a new account that was added later.

I hope that makes sense.

**Comment by kpachhai** [comment]
Any answer to this would be appreciated. We're also running into the same issue @Montoya
